extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -223,11 +223,12 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 				await this.bookTocManager.recovery();
 				vscode.window.showErrorMessage(loc.editBookError(updateBook.book.contentPath, e instanceof Error ? e.message : e));
 			} finally {
-				this.fireBookRefresh(targetBook);
-				if (sourceBook) {
+				await targetBook.initializeContents();
+				if (sourceBook && sourceBook.bookPath !== targetBook.bookPath) {
 					// refresh source book model to pick up latest changes
-					this.fireBookRefresh(sourceBook);
+					await sourceBook.initializeContents();
 				}
+				this._onDidChangeTreeData.fire(undefined);
```

Comment:
May want to put this in a finally in case the source book initialize has an error

Reply:
This is already inside finally, do you mean putting it in a new try/catch?

Reply:
Yes. If the second initializeBookContents call throws then this will never fire the event - even if the first completed successfully.

Although if you think that purposely only sending the event if both complete is correct that's fine. Just want to make sure you're handling it correctly.

Reply:
I added a try/finally inside the finally